mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_external_access_rule_test.go

```diff
@@ -0,0 +1,203 @@
+  parent =  google_vmwareengine_network_policy.external-access-rule-np.id
+  description = "description1"
+  priority = "101"
+  action = "ALLOW"
+  ip_protocol = "TCP"
+  source_ip_ranges {
+    ip_address_range = "0.0.0.0/0"
+  }
+  source_ports = ["80"]
+  destination_ip_ranges {
+    external_address = google_vmwareengine_external_address.external-access-rule-ea.id
+  }
+  destination_ports = ["433"]
+  depends_on = [
+    google_vmwareengine_external_address.external-access-rule-ea,
+  ]
+}
+
+data "google_vmwareengine_external_access_rule" "ds" {
+  name = google_vmwareengine_external_access_rule.vmw-engine-external-access-rule.name
+  parent = google_vmwareengine_network_policy.external-access-rule-np.id
+  depends_on = [
+    google_vmwareengine_external_access_rule.vmw-engine-external-access-rule,
+  ]
+}
+`, context)
+}
+
+func testVmwareEngineExternalAccessRuleUpdateConfig(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+
+resource "google_vmwareengine_network" "external-access-rule-nw" {
+  name        = "tf-test-sample-external-access-rule-nw%{random_suffix}"
+  location    = "global"
+  type        = "STANDARD"
+}
+
+resource "google_vmwareengine_private_cloud" "external-access-rule-pc" {
+  location    = "%{region}-a"
+  name        = "tf-test-sample-external-access-rule-pc%{random_suffix}"
+  network_config {
+    management_cidr       = "192.168.1.0/24"
+    vmware_engine_network = google_vmwareengine_network.external-access-rule-nw.id
+  }
+
+  management_cluster {
+    cluster_id = "tf-test-sample-external-access-rule-cluster%{random_suffix}"
+    node_type_configs {
+      node_type_id = "standard-72"
+      node_count   = 3
+    }
+  }
+}
+
+resource "google_vmwareengine_network_policy" "external-access-rule-np" {
+  location = "%{region}"
+  name = "tf-test-sample-external-access-rule-np%{random_suffix}"
+  edge_services_cidr = "192.168.0.0/26"
+  vmware_engine_network = google_vmwareengine_network.external-access-rule-nw.id
+	internet_access {
+    enabled = true
+  }
+  external_ip {
+    enabled = true
+  }
+}
+
+resource "google_vmwareengine_external_address" "external-access-rule-ea" {
+  name = "tf-test-sample-external-access-rule-ea%{random_suffix}"
+  parent =  google_vmwareengine_private_cloud.external-access-rule-pc.id
+  internal_ip = "192.168.0.73"
+  description = "Sample description."
+  depends_on = [
+    google_vmwareengine_network_policy.external-access-rule-np,
+  ]
+}
+
+resource "google_vmwareengine_external_access_rule" "vmw-engine-external-access-rule" {
+  name = "tf-test-sample-external-access-rule%{random_suffix}"
+  parent =  google_vmwareengine_network_policy.external-access-rule-np.id
+  description = "description2"
+  priority = "102"
+  action = "DENY"
+  ip_protocol = "UDP"
+  source_ip_ranges {
+    ip_address = "192.168.40.0"
+  }
+  source_ports = ["81"]
+  destination_ip_ranges {
+    ip_address_range = "0.0.0.0/0"
+  }
+  destination_ports = ["0-65535"]
```

Comment:
Could you please update this test to include multiple entries for `source_ip_ranges`, `source_ports`, `destination_ip_ranges`, and `destination_ports`?

We'd want to identify whether the API can reorder the items in those arrays to avoid situations where a user's config lists things out-of-order and the API returns them reordered, causing a permanent diff detected by Terraform. If the tests help us identify an issue like that we can [change the fields to be Sets instead](https://github.com/hashicorp/terraform-provider-google/issues/16771).

Reply:
Makes sense, thanks for the feedback. I have modified the test to include multiple entries for all the mentioned fields. The modified test passes locally as well.